src/test/scala/com/lightbend/kafkalagexporter/PrometheusEndpointSinkTest.scala

```diff
@@ -45,38 +45,60 @@ class PrometheusEndpointSinkTest extends fixture.FreeSpec with Matchers {
         Set("kafka_consumergroup_group_max_lag", "kafka_consumergroup_group_max_lag_seconds")
     }
 
-//    "append global labels to metric labels" in { fixture =>
-//      val groupLabel = Map(
-//        "cluster" -> Map(
-//          "environment" ->"dev",
-//          "org" -> "organization",
-//        )
-//      )
-//      val sink = PrometheusEndpointSink(Metrics.definitions, List(".*"), groupLabel, fixture.server, fixture.registry)
-//      sink.report(Metrics.GroupValueMessage(Metrics.MaxGroupTimeLagMetric, "cluster", "group", 1))
-//
-//      val metricSamples = fixture.registry.metricFamilySamples().asScala.toList
-//      val maxGroupTimeLagMetricSamples = metricSamples.filter(_.name.equals(Metrics.MaxGroupTimeLagMetric.name)).flatMap(_.samples.asScala)
-//
-//      maxGroupTimeLagMetricSamples should have length 1
-//      val labels = maxGroupTimeLagMetricSamples.flatMap(_.labelNames.asScala)
-//      val labelValues = maxGroupTimeLagMetricSamples.flatMap(_.labelValues.asScala)
-//      (labels zip labelValues).toMap should contain theSameElementsAs
-//        Map(
-//          "environment" ->"dev",
-//          "org" -> "organization",
-//          "cluster_name" -> "cluster",
-//          "group" -> "group",
-//        )
-//
-//      sink.remove(Metrics.GroupRemoveMetricMessage(Metrics.MaxGroupTimeLagMetric, "cluster", "group"))
-//
-//      val metricSamplesAfterRemoval = fixture.registry.metricFamilySamples().asScala.toList
-//      val maxGroupTimeLagMetricSamplesAfterRemoval = metricSamplesAfterRemoval.filter(_.name.equals(Metrics.MaxGroupTimeLagMetric.name)).flatMap(_.samples.asScala)
-//
-//
-//      maxGroupTimeLagMetricSamplesAfterRemoval should have length 0
-//    }
+    "append global labels to metric labels" in { fixture =>
+      val groupLabel = Map(
+        "cluster" -> Map(
+          "environment" -> "dev",
+          "org" -> "organization",
+          "location" -> ""
+        ),
+        "cluster2" -> Map(
+          "environment" -> "prod",
+          "org" -> "organization2",
+          "location" -> "canada"
+        )
```

Comment:
It's possible for clusters with no labels defined to report metrics.  When you use the Strimzi cluster watcher, clusters are added at runtime and will have an empty set of labels.  Can you add/update a test to include a cluster with labels defined, and one with none defined, and assert that it works.  I assume the result would be blank label values for the clusters that have none defined.

Reply:
Nice catch :) I added those test cases and made some minor changes in the implementation to accommodate this requirement.